src/util/ExpressionUtils.cpp

```diff
@@ -16,6 +16,141 @@
 namespace nebula {
 namespace graph {
 
+// null for not found
+const Expression *ExpressionUtils::findAny(const Expression *self,
+                                           const std::unordered_set<Expression::Kind> &expected) {
+    FindAnyExprVisitor visitor(expected);
+    const_cast<Expression *>(self)->accept(&visitor);
+    return visitor.expr();
+}
+
+bool ExpressionUtils::hasAny(const Expression *expr,
```

Comment:
Better put the short function in header file for inline.

Reply:
ACK.